hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsMultipleNodeTest.java

```diff
@@ -52,8 +51,27 @@
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class LocalMapStatsMultipleNodeTest extends HazelcastTestSupport {
 
+    //https://github.com/hazelcast/hazelcast/issues/21785
     @Test
-    public void testHits_whenMultipleNodes() throws InterruptedException {
+    public void testHits_isZero_afterPut() {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
+        HazelcastInstance instance1 = factory.newHazelcastInstance(getConfig());
+        IMap<Integer, Integer> iMap = instance1.getMap("test-hits");
+
+        for (int i = 0; i < 100; i++) {
+            iMap.put(i, i);
+        }
+
+        HazelcastInstance instance2 = factory.newHazelcastInstance(getConfig());
+        HazelcastInstance instance3 = factory.newHazelcastInstance(getConfig());
+        waitAllForSafeState(instance1, instance2, instance3);
+        assertEquals(0, instance1.getMap("test-hits").getLocalMapStats().getHits());
+        assertEquals(0, instance2.getMap("test-hits").getLocalMapStats().getHits());
+        assertEquals(0, instance3.getMap("test-hits").getLocalMapStats().getHits());
```

Comment:
Good have this test since putting a new entry should not generate hits, only updates and gets should generate it.
As far as i see, the main issue which makes hits < 0 is updating hits during replication. Removing that update should be sufficient to fix the issue behind this PR. 

@arodionov how does this sound?

Reply:
Is it that update?
https://github.com/arodionov/hazelcast/blob/e6a706fced9d3e69ee75fe5039e862d7f1ded325/hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java#L234

Reply:
yes, it should be that.

Reply:
I think in that case the enterprise version should also update https://github.com/hazelcast/hazelcast-enterprise/blob/master/hazelcast-enterprise/src/main/java/com/hazelcast/map/impl/recordstore/EnterpriseTieredRecordStore.java#L161